waspc/data/Generator/templates/server/src/routes/auth/passport/google/google.js

```diff
@@ -0,0 +1,76 @@
+import express from 'express'
+import passport from 'passport'
+import GoogleStrategy from 'passport-google-oauth20'
+
+import waspServerConfig from '../../../../config.js'
+import { contextWithUserEntity, authConfig, findOrCreateUserByExternalAuthAssociation } from '../../utils.js'
+import { sign } from '../../../../core/auth.js'
+import { configFn, getUserFields } from './googleConfig.js'
+
+const config = ((config) => {
+  if (!config?.clientId) {
+    throw new Error("auth.google.configFn must return an object with a clientId property.")
+  }
+
+  if (!config?.clientSecret) {
+    throw new Error("auth.google.configFn must return an object with a clientSecret property.")
+  }
+
+  if (!config?.scope) {
+    throw new Error("auth.google.configFn must return an object with a scope property.")
+  } else if (!Array.isArray(config.scope) || !config.scope.includes('profile')) {
+    throw new Error("auth.google.configFn returned an object with an invalid scope property. It must be an array including 'profile'.")
+  }
+
+  return config
+})(configFn())
+
+passport.use('waspGoogleLoginStrategy', new GoogleStrategy({
+  clientID: config.clientId,
+  clientSecret: config.clientSecret,
+  callbackURL: `${waspServerConfig.frontendUrl}/auth/login/google`,
```

Comment:
Changed callback URL and names in here to be "login" oriented, as in the future we may have slight variants that can handle associating an existing username with an external auth account.